src/ceph-daemon/ceph-daemon

```diff
@@ -1060,7 +1112,7 @@ def command_bootstrap():
         f.write(config)
 
     mon_c = get_container(fsid, 'mon', mon_id)
-    deploy_daemon_units(fsid, uid, gid, 'mon', mon_id, mon_c)
+    deploy_daemon(fsid, 'mon', mon_id, mon_c, uid, gid, None, None)
```

Comment:
fixed i think?

Reply:
mypy still doesn't like it, but I think that's valid ... otherwise we attempt to write a config/keyring file using NoneType:
https://github.com/ceph/ceph/blob/9f1fee0e8154e03cfa4a7f653e11e0bb21061b06/src/ceph-daemon/ceph-daemon#L575


Maybe just raise a RuntimeError in this case?
```
diff --git a/src/ceph-daemon/ceph-daemon b/src/ceph-daemon/ceph-daemon
index 1894776244..ed4fc53dca 100755
--- a/src/ceph-daemon/ceph-daemon
+++ b/src/ceph-daemon/ceph-daemon
@@ -564,10 +564,15 @@ def extract_uid_gid():
     return (int(uid), int(gid))
 
 def deploy_daemon(fsid, daemon_type, daemon_id, c, uid, gid,
-                  config, keyring):
+                  config=None, keyring=None):
     # type: (str, str, Union[int, str], CephContainer, int, int, str, str) -> None
     if daemon_type == 'mon' and not os.path.exists(get_data_dir(fsid, 'mon',
                                                                 daemon_id)):
+        if not config:
+            raise RuntimeError('config cannot be empty!')
+        if not keyring:
+            raise RuntimeError('keyring cannot be empty!')
+
         # tmp keyring file
         tmp_keyring = tempfile.NamedTemporaryFile(mode='w')
         os.fchmod(tmp_keyring.fileno(), 0o600)
@@ -1112,8 +1117,7 @@ def command_bootstrap():
         f.write(config)
 
     mon_c = get_container(fsid, 'mon', mon_id)
-    deploy_daemon(fsid, 'mon', mon_id, mon_c, uid, gid,
-                  config=None, keyring=None)
+    deploy_daemon(fsid, 'mon', mon_id, mon_c, uid, gid)
 
     # client.admin key + config to issue various CLI commands
     tmp_admin_keyring = tempfile.NamedTemporaryFile(mode='w')

```